workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/ScreenComposableFactoryFinder.kt

```diff
@@ -22,27 +24,32 @@ public interface ScreenComposableFactoryFinder {
     @Suppress("UNCHECKED_CAST")
     return factoryOrNull
       ?: (rendering as? ComposeScreen)?.let {
-        ScreenComposableFactory<ComposeScreen> { rendering, environment ->
-          rendering.Content(environment)
+        ScreenComposableFactory<ComposeScreen> { rendering ->
+          rendering.Content()
         } as ScreenComposableFactory<ScreenT>
       }
 
       // Support for Compose BackStackScreen, BodyAndOverlaysScreen treatments would go here,
       // if it were planned. See similar blocks in ScreenViewFactoryFinder
 
       ?: (rendering as? NamedScreen<*>)?.let {
-        ScreenComposableFactory<NamedScreen<*>> { rendering, environment ->
-          val innerFactory = rendering.content.toComposableFactory(environment)
-          innerFactory.Content(rendering.content, environment)
-          // WorkflowRendering(rendering.content, environment)
+        ScreenComposableFactory<NamedScreen<*>> { rendering ->
+          val innerFactory = rendering.content
+            .toComposableFactory(LocalWorkflowEnvironment.current)
+          innerFactory.Content(rendering.content)
         } as ScreenComposableFactory<ScreenT>
       }
       ?: (rendering as? EnvironmentScreen<*>)?.let {
-        ScreenComposableFactory<EnvironmentScreen<*>> { rendering, environment ->
-          val comboEnv = environment + rendering.environment
-          val innerFactory = rendering.content.toComposableFactory(comboEnv)
-          innerFactory.Content(rendering.content, comboEnv)
-          // WorkflowRendering(rendering.content, comboEnv)
+        ScreenComposableFactory<EnvironmentScreen<*>> { rendering ->
+          val currentEnv = LocalWorkflowEnvironment.current
+          val innerFactory = rendering.content.toComposableFactory(
+            currentEnv + rendering.environment
+          )
+
+          val comboEnv = remember(currentEnv) { currentEnv + rendering.environment }
```

Comment:
Should rendering.environment be included as a key in the remember function to ensure comboEnv is recomputed when it changes? Currently, it's only keyed on currentEnv, which may not cover all use cases.

Reply:
Yes, it can change.